recipe/meta.yaml

```
@@ -1,44 +1,70 @@
+{% set name = "uvicorn" %}
 {% set version = "0.16.0" %}
 
 package:
-  name: uvicorn
+  name: {{ name }}-recipe
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/u/uvicorn/uvicorn-{{ version }}.tar.gz
   sha256: eacb66afa65e0648fcbce5e746b135d09722231ffffc61883d4fac2b62fbea8d
 
 build:
-  number: 0
-  script: {{ PYTHON }} -m pip install . -vv
-  entry_points:
-    - uvicorn = uvicorn.main:main
-
-requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-  host:
-    - python
-    - pip
-  run:
-    - python
-    - asgiref >=3.4.0
-    - click >=7.*
-    - h11 >=0.8
-    - typing-extensions  # [py<38]
+  number: 1
 
 test:
   requires:
     - pip
-  imports:
-    - uvicorn
-  commands:
-    - pip check
-    - uvicorn --help
+
+outputs:
+  - name: {{ name }}
+
+    build:
+      entry_points:
+        - uvicorn = uvicorn.main:main
+    script: uvicorn_build.bat  # [win]
+    script: uvicorn_build.sh   # [not win]
+
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+      host:
+        - python
+        - pip
+      run:
+        - python
+        - asgiref >=3.4.0
+        - click >=7.*
+        - h11 >=0.8
+        - typing-extensions  # [py<38]
+
+    test:
+      imports:
+        - uvicorn
+      commands:
+        - pip check
+        - uvicorn --help
+
+  - name: {{ name }}-standard
+
+    requirements:
```

Also needs to be in host for file deduplication to work
```suggestion
    requirements:
      host:
        - python
        - pip
        - {{ pin_subpackage('uvicorn', exact=True) }}
```